### ISSUE TITLE
Fedora 23 Support

### ISSUE THREAD

**Issue by Jeryia** [arescentral/antares#164]
This is a request to add support for Fedora 23 linux. 

Issues preventing this from working:
- Initially the ./configure script will complain that ninja is missing. Doing some digging I've found that Fedora names the binary for ninja 'ninja-build'. A change to the ./configure and Makefile to look for ninja-build instead of ninja seems to fix the issue. Could likely add something to the ./configure script to check for that binary.
- Secondary issue comes with compiling.  I'm not quite sure what's going on there:

    $ make
    ninja: Entering directory `out/cur'
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.print.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.print.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/print.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.print.o
    In file included from ../../ext/libsfz/src/all/sfz/print.cpp:9:
    In file included from ../../ext/libsfz/include/all/sfz/format.hpp:11:
    ../../ext/libsfz/include/all/sfz/string.hpp:11:10: fatal error: 'cstddef' file not found
    #include <cstddef>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.bytes.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.bytes.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/bytes.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.bytes.o
    In file included from ../../ext/libsfz/src/all/sfz/bytes.cpp:6:
    ../../ext/libsfz/include/all/sfz/bytes.hpp:11:10: fatal error: 'cstddef' file not found
    #include <cstddef>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.encoding.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.encoding.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/encoding.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.encoding.o
    ../../ext/libsfz/src/all/sfz/encoding.cpp:8:10: fatal error: 'algorithm' file not found
    #include <algorithm>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.args.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.args.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/args.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.args.o
    In file included from ../../ext/libsfz/src/all/sfz/args.cpp:6:
    ../../ext/libsfz/include/all/sfz/args.hpp:9:10: fatal error: 'limits' file not found
    #include <limits>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.digest.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.digest.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/digest.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.digest.o
    ../../ext/libsfz/src/all/sfz/digest.cpp:9:10: fatal error: 'limits' file not found
    #include <limits>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.format.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.format.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/format.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.format.o
    In file included from ../../ext/libsfz/src/all/sfz/format.cpp:6:
    In file included from ../../ext/libsfz/include/all/sfz/format.hpp:11:
    ../../ext/libsfz/include/all/sfz/string.hpp:11:10: fatal error: 'cstddef' file not found
    #include <cstddef>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.exception.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.exception.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/exception.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.exception.o
    In file included from ../../ext/libsfz/src/all/sfz/exception.cpp:6:
    ../../ext/libsfz/include/all/sfz/exception.hpp:9:10: fatal error: 'exception' file not found
    #include <exception>
             ^
    1 error generated.
    [10/233] CXX obj/src/linux/libantares-linux.http.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/src/linux/libantares-linux.http.o.d -DNDEBUG -I../../include -Iobj/libantares-linux.gen/include -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../src/linux/http.cpp -o obj/src/linux/libantares-linux.http.o
    In file included from ../../src/linux/http.cpp:19:
    In file included from ../../include/net/http.hpp:22:
    In file included from ../../ext/libsfz/include/all/sfz/sfz.hpp:10:
    ../../ext/libsfz/include/all/sfz/args.hpp:9:10: fatal error: 'limits' file not found
    #include <limits>
             ^
    1 error generated.
    [10/233] CXX obj/ext/libsfz/src/all/sfz/libsfz.json.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/ext/libsfz/src/all/sfz/libsfz.json.o.d -DNDEBUG -I../../ext/libsfz/src/all -I../../ext/libsfz/src/posix -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../ext/libsfz/src/all/sfz/json.cpp -o obj/ext/libsfz/src/all/sfz/libsfz.json.o
    In file included from ../../ext/libsfz/src/all/sfz/json.cpp:6:
    ../../ext/libsfz/include/all/sfz/json.hpp:9:10: fatal error: 'map' file not found
    #include <map>
             ^
    1 error generated.
    ninja: build stopped: subcommand failed.
    Makefile:6: recipe for target 'all' failed
    make: *** [all] Error 1

I'd be happy to assist with any testing you may wish to do. 

Thanks for your work on Antares! We all appreciate your contributions to the open source community!
Once this is working, if you want someone to make an rpm for you for easy installation on Fedora and other rpm based systems, I'd be happy to create it for this project.

**Comment by sfiera** [comment]
I'm away from a computer this weekend so I won't be able to make any
changes immediately, but it looks like you're missing a standard C++
library. I think the check for libc++ isn't good enough. Can you track down
Fedora's libc++ and compile again?

**Comment by sfiera** [comment]
I pushed some improvements to ./configure to head. Should work better for
you. The Makefile won't work properly, but you can run ninja-build manually
after configuring.

**Comment by Jeryia** [comment]
Did a bit more digging, looks like another naming issue. I could not find any references to libc++ in the fedora repositories. However, I did find that libstdc++-devel. 

After trying that I replaced all references of libc++ with libstdc++. This seems to satisfy the new c++ library checking in the new configure script (thank you for the changes!). I then make the same change to the ninja line in the Makefile and ran make again to the following results:

    $ make
    ninja: Entering directory `out/cur'
    [30/348] CXX obj/src/math/libantares-math.rotation.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/src/math/libantares-math.rotation.o.d -DNDEBUG -I../../include -Iobj/libantares-math.gen/include -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libstdc++  -c ../../src/math/rotation.cpp -o obj/src/math/libantares-math.rotation.o
    In file included from ../../src/math/rotation.cpp:24:
    In file included from ../../include/game/sys.hpp:25:
    In file included from ../../include/drawing/sprite-handling.hpp:23:
    ../../include/drawing/color.hpp:109:12: error: use of undeclared identifier 'memcmp'; did you mean  'wmemcmp'?
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) == 0;
               ^~~~~~
               wmemcmp
    /usr/include/wchar.h:327:12: note: 'wmemcmp' declared here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
               ^
    In file included from ../../src/math/rotation.cpp:24:
    In file included from ../../include/game/sys.hpp:25:
    In file included from ../../include/drawing/sprite-handling.hpp:23:
    ../../include/drawing/color.hpp:109:19: error: cannot initialize a parameter of type 'const wchar_t *' with an rvalue of type 'const antares::RgbColor *'
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) == 0;
                      ^~~~
    /usr/include/wchar.h:327:36: note: passing argument to parameter '__s1' here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
                                       ^
    In file included from ../../src/math/rotation.cpp:24:
    In file included from ../../include/game/sys.hpp:25:
    In file included from ../../include/drawing/sprite-handling.hpp:23:
    ../../include/drawing/color.hpp:113:12: error: use of undeclared identifier 'memcmp'; did you mean 'wmemcmp'?
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) != 0;
               ^~~~~~
               wmemcmp
    /usr/include/wchar.h:327:12: note: 'wmemcmp' declared here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
               ^
    In file included from ../../src/math/rotation.cpp:24:
    In file included from ../../include/game/sys.hpp:25:
    In file included from ../../include/drawing/sprite-handling.hpp:23:
    ../../include/drawing/color.hpp:113:19: error: cannot initialize a parameter of type 'const wchar_t *' with an  rvalue of type 'const antares::RgbColor *'
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) != 0;
                      ^~~~
    /usr/include/wchar.h:327:36: note: passing argument to parameter '__s1' here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
                                       ^
    4 errors generated.
    [30/348] CXX obj/src/math/libantares-math.random.o
    FAILED: /usr/bin/clang++ -MMD -MF obj/src/math/libantares-math.random.o.d -DNDEBUG -I../../include -Iobj/libantares-math.gen/include -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libstdc++  -c ../../src/math/random.cpp -o obj/src/math/libantares-math.random.o
    In file included from ../../src/math/random.cpp:22:
    In file included from ../../include/game/globals.hpp:29:
    ../../include/drawing/color.hpp:109:12: error: use of undeclared identifier 'memcmp'; did you mean  'wmemcmp'?
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) == 0;
               ^~~~~~
               wmemcmp
    /usr/include/wchar.h:327:12: note: 'wmemcmp' declared here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
               ^
    In file included from ../../src/math/random.cpp:22:
    In file included from ../../include/game/globals.hpp:29:
    ../../include/drawing/color.hpp:109:19: error: cannot initialize a parameter of type 'const wchar_t *' with an rvalue of type 'const antares::RgbColor *'
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) == 0;
                      ^~~~
    /usr/include/wchar.h:327:36: note: passing argument to parameter '__s1' here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
                                       ^
    In file included from ../../src/math/random.cpp:22:
    In file included from ../../include/game/globals.hpp:29:
    ../../include/drawing/color.hpp:113:12: error: use of undeclared identifier 'memcmp'; did you mean 'wmemcmp'?
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) != 0;
               ^~~~~~
               wmemcmp
    /usr/include/wchar.h:327:12: note: 'wmemcmp' declared here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
               ^
    In file included from ../../src/math/random.cpp:22:
    In file included from ../../include/game/globals.hpp:29:
    ../../include/drawing/color.hpp:113:19: error: cannot initialize a parameter of type 'const wchar_t *' with an rvalue of type 'const antares::RgbColor *'
        return memcmp(&lhs, &rhs, sizeof(RgbColor)) != 0;
                      ^~~~
    /usr/include/wchar.h:327:36: note: passing argument to parameter '__s1' here
    extern int wmemcmp (const wchar_t *__s1, const wchar_t *__s2, size_t __n)
                                   ^
    4 errors generated.
    [30/348] CXX obj/ext/libsfz/src/posix/sfz/libsfz.os.o
    ninja: build stopped: subcommand failed.
    Makefile:6: recipe for target 'all' failed
    make: *** [all] Error 1


It looks like a step forward, though definitely let me know if I'm on the wrong track. My knowledge of c++ is very limited. 

The inner C programmer in me wants to add '#include <string.h>' to the mentioned files; However, I expect there's likely a more fundamental problem going on otherwise this wouldn't compile elsewhere. Your thoughts?

**Comment by sfiera** [comment]
Where does memcmp come from on your system? (check `man memcmp`) If I
recall, it's from string.h on my system, but it's possible that either:

1. It's somewhere else for you
2. I've included something that includes string.h transitively on my
system, but doesn't for you, and forgotten to include string.h.

I apologize if it's the latter. Go ahead and add all the includes you want.

At the time I added the libc++ requirement, libstdc++ didn't yet support
recent C++ features, but it's possible that it's caught up. You can see how
far it gets you :)

**Comment by sfiera** [comment]
Looks like I did miss it. Added at HEAD.

**Comment by Jeryia** [comment]
Howdy,
Actually doing more research, I think this is a flaw in Fedora not your code. Apparently Fedora does not package libc++, and only offers libstdc++. These libraries do not offer the same functionality. (putting some includes in does get me farther, however, I ran into make issues on functions offered only by libc++ (such as push_heap).

Will try a source install of libc++ at some point, but my first shot at it failed. Unless you're willing to port this to llibstdc++ I doubt we can really say this is works on Fedora. Thanks for the help :).

**Comment by sfiera** [comment]
Really? That's odd; push_heap isn't a new function. Unfortunately, I don't think I can do much to help. Apple stopped packaging recent versions of libstdc++ because of GPL3, so I don't have a version with recent enough features to port to.

**Comment by Jeryia** [comment]
understood, 
I appreciate the help. Perhaps one day Fedora will offer libc++. It looks like many have wanted it. I'm actually quite surprised that not even the supplementary repos offer it like rpmfusion. 

Definitely let me know if you think of something for me to try. Otherwise, I guess there's not much that can be done.

**Comment by Mailaender** [comment]
Try installing `llvm-devel` on Fedora which should include http://libcxx.llvm.org/ at least according to https://pkgs.org/fedora-23/fedora-i386/llvm-devel-3.7.0-1.fc23.i686.rpm.html

See also my .spec file I created for @openSUSE: https://build.opensuse.org/request/show/391437

**Comment by susnux** [comment]
The errors about missing `memcmp`, `memcpy` and `memset` are shown due to the fact that `#include <cstring>` is missing.
This happens with libstdc++ because libc++ adds that header somewhere in the backend...
Same with `push_heap`. It is defined in `#include <algorithm>`

**Comment by Mailaender** [comment]
Awesome, thanks! See the patch in action at: https://build.opensuse.org/request/show/393352

**Comment by sfiera** [comment]
Now that I've updated my Ubuntu machine to 16.04, I was able to try a more recent version of libstdc++. I've updated master to not specify a -stdlib on Linux, and added missing headers. If that works for you, I'll cut a release soon.

**Comment by Jeryia** [comment]
I'm on work travel right now, (and my laptop's clang seems messed up somehow). I'll give these a shot when I get back. A side note I did try the llvm-devel without any luck. I can post the exact error message when I get back

**Comment by Jeryia** [comment]
Hmm,
Still doesn't seem to work. libstdc++ is installed, but I'm getting this error when running make:
ninja: Entering directory `out/cur'
[10/370] CXX obj/src/config/libantares-config.preferences.o
FAILED: /usr/bin/clang++ -MMD -MF obj/src/config/libantares-config.preferences.o.d -DNDEBUG -I../../include -Iobj/libantares-config.gen/include -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libc++  -c ../../src/config/preferences.cpp -o obj/src/config/libantares-config.preferences.o
In file included from ../../src/config/preferences.cpp:19:
In file included from ../../include/config/preferences.hpp:22:
In file included from ../../ext/libsfz/include/all/sfz/sfz.hpp:10:
../../ext/libsfz/include/all/sfz/args.hpp:9:10: fatal error: 'limits' file not found
#include <limits>
         ^
1 error generated.
[10/370] CC obj/ext/libsndfile-gyp/libsndfile-1.0.25/src/libsndfile-common.aiff.o
ninja: build stopped: subcommand failed.
Makefile:6: recipe for target 'all' failed
make: *** [all] Error 1



very strange as that should be in the standard library. I'll see what I can find.

**Comment by sfiera** [comment]
-stdlib=libc++ is still in your command-line there. Did you re-run ./configure?

I think <limits> just happens to be the first stdlib header you're hitting, but any other stdlib header would fail too.

**Comment by Jeryia** [comment]
Yeah, I pulled down a fresh copy from git, and ran configure.
Ok, I've done a find/replace on libc++ to libstdc++. 

New errors:
ninja: Entering directory `out/cur'
[42/370] CC obj/ext/glfw/glfw-3.1.1/src/libglfw.glx_context.o
../../ext/glfw/glfw-3.1.1/src/glx_context.c: In function ‘_glfwCreateContext’:
../../ext/glfw/glfw-3.1.1/src/glx_context.c:155:12: warning: ‘native’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return glXCreateNewContext(_glfw.x11.display,
            ^
../../ext/glfw/glfw-3.1.1/src/glx_context.c:294:17: note: ‘native’ was declared here
     GLXFBConfig native;
                 ^
[152/370] CXX obj/src/game/libantares-game.cheat.o
FAILED: /usr/bin/clang++ -MMD -MF obj/src/game/libantares-game.cheat.o.d -DNDEBUG -I../../include -Iobj/libantares-game.gen/include -I../../ext/libsfz/include/all -I../../ext/libsfz/include/posix -Os -std=c++11 -stdlib=libstdc++  -c ../../src/game/cheat.cpp -o obj/src/game/libantares-game.cheat.o
../../src/game/cheat.cpp:61:15: error: no matching function for call to 'find'
    auto it = std::find(sys.cheat.codes.begin(), sys.cheat.codes.end(), code_string);
              ^~~~~~~~~
/usr/bin/../lib/gcc/x86_64-redhat-linux/5.3.1/../../../../include/c++/5.3.1/bits/streambuf_iterator.h:369:5: note: candidate template ignored: could not match 'istreambuf_iterator' against '__normal_iterator'
    find(istreambuf_iterator<_CharT> __first,
    ^
1 error generated.

**Comment by sfiera** [comment]
Oops. I pushed to sfiera/antares but forgot to push the same change to arescentral/antares. Updated now.

**Comment by Jeryia** [comment]
It compiles now :+1: 
However, it core dumps when trying to run. It does show one warning when compiling (don't know if it's an issue)
$ make
ninja: Entering directory `out/cur'
[265/370] CC obj/ext/glfw/glfw-3.1.1/src/libglfw.glx_context.o
../../ext/glfw/glfw-3.1.1/src/glx_context.c: In function ‘_glfwCreateContext’:
../../ext/glfw/glfw-3.1.1/src/glx_context.c:155:12: warning: ‘native’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     return glXCreateNewContext(_glfw.x11.display,
            ^
../../ext/glfw/glfw-3.1.1/src/glx_context.c:294:17: note: ‘native’ was declared here
     GLXFBConfig native;
                 ^
[370/370] LINK object-data


$ out/cur/antares-install-data
/home/#####/.local/share/antares/scenarios is up-to-date!

$ ./out/cur/antares
terminate called after throwing an instance of 'sfz::Exception'
  what():  glfwInit()
Aborted (core dumped)

I can attach the core file if it would be helpfull.

**Comment by Jeryia** [comment]
Ran the make test output. Based on the output I fear we may not be as close as I had hoped:
$ make test
ninja: Entering directory `out/cur'
ninja: no work to do.
scripts/test.py
Running 25 tests:
shapes failed:
====================
diff failed:
diff: test/shapes: No such file or directory

====================
mission-briefing failed:
====================
diff failed:
diff: test/mission-briefing: No such file or directory

====================
astrotrash-plus failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/astrotrash-plus.NLRP: 0

====================
blood-toil-tears-sweat failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/blood-toil-tears-sweat.NLRP: 0

====================
hand-over-fist failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/hand-over-fist.NLRP: 0

====================
hornets-nest failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/hornets-nest.NLRP: 0

====================
make-way failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/make-way.NLRP: 0

====================
moons-for-goons failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/moons-for-goons.NLRP: 0

====================
out-of-the-frying-pan failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/out-of-the-frying-pan.NLRP: 0

====================
shoplifter-1 failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/shoplifter-1.NLRP: 0

====================
space-race failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/space-race.NLRP: 0

====================
the-left-hand failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/the-left-hand.NLRP: 0

====================
the-mothership-connection failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/the-mothership-connection.NLRP: 0

====================
the-stars-have-ears failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/the-stars-have-ears.NLRP: 0

====================
while-the-iron-is-hot failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/while-the-iron-is-hot.NLRP: 0

====================
yo-ho-ho failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/yo-ho-ho.NLRP: 0

====================
you-should-have-seen-the-one-that-got-away failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

====================
pause failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

====================
tint failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

diff failed:
diff: test/tint: No such file or directory

====================
main-screen failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

diff failed:
diff: test/tint: No such file or directory

offscreen failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

====================
object-data failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

diff failed:
diff: test/tint: No such file or directory

offscreen failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

diff failed:
diff: test/object-data: No such file or directory

====================
and-it-feels-so-good failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

diff failed:
diff: test/tint: No such file or directory

offscreen failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

diff failed:
diff: test/object-data: No such file or directory

replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/and-it-feels-so-good.NLRP: 0

====================
build-pix failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

diff failed:
diff: test/tint: No such file or directory

offscreen failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

diff failed:
diff: test/object-data: No such file or directory

replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/and-it-feels-so-good.NLRP: 0

build-pix failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

====================
options failed:
====================
replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/you-should-have-seen-the-one-that-got-away.NLRP: 0

diff failed:
diff: test/pause: No such file or directory

diff failed:
diff: test/tint: No such file or directory

offscreen failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

diff failed:
diff: test/object-data: No such file or directory

replay failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  test/and-it-feels-so-good.NLRP: 0

build-pix failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

offscreen failed:
terminate called after throwing an instance of 'sfz::Exception'
  what():  glXChooseFBConfig() was null

====================
  fixed-test                               PASSED in 0.02s
  shapes                                   FAILED in 0.05s
  mission-briefing                         FAILED in 0.20s
  astrotrash-plus                          FAILED in 0.01s
  blood-toil-tears-sweat                   FAILED in 0.01s
  hand-over-fist                           FAILED in 0.01s
  hornets-nest                             FAILED in 0.01s
  make-way                                 FAILED in 0.01s
  moons-for-goons                          FAILED in 0.01s
  out-of-the-frying-pan                    FAILED in 0.01s
  shoplifter-1                             FAILED in 0.01s
  space-race                               FAILED in 0.01s
  the-left-hand                            FAILED in 0.01s
  the-mothership-connection                FAILED in 0.01s
  the-stars-have-ears                      FAILED in 0.01s
  while-the-iron-is-hot                    FAILED in 0.01s
  yo-ho-ho                                 FAILED in 0.01s
  you-should-have-seen-the-one-that...     FAILED in 0.01s
  pause                                    FAILED in 0.47s
  tint                                     FAILED in 0.52s
  main-screen                              FAILED in 0.56s
  object-data                              FAILED in 0.57s
  and-it-feels-so-good                     FAILED in 0.52s
  build-pix                                FAILED in 1.06s
  options                                  FAILED in 1.05s

Ran 25 tests in 1.07s
24 tests failed.
Makefile:9: recipe for target 'test' failed
make: *** [test] Error 1
[grahamvh@kosh antares]$ make run
ninja: Entering directory `out/cur'
ninja: no work to do.
terminate called after throwing an instance of 'sfz::Exception'
  what():  glfwInit()
/bin/sh: line 1:  8933 Aborted                 (core dumped) out/cur/antares

**Comment by sfiera** [comment]
Do you have the `test` submodule? The default instructions avoid pulling it down, because it's not necessary to build and run the game, and quadruples the size of the repo. If not, just run `git submodule update`.

I don't know what to do about glfwInit() or glXChooseFBConfig() failing. There's a good chance I'm misusing the latter, but glfwInit() is a library function with no arguments and no setup. Just to ask the obvious questions, you're using a machine with basic things like a graphics card and a monitor, right? :)

Assuming that you have the `test` submodule but OpenGL isn't working for you, the tests build-pix, options, and main-screen will fail but everything else should pass.

**Comment by Jeryia** [comment]
Not sure what I was doing before, but it seems to be working now. Pulling in the submodules allowed all of the tests to pass. 

Two notes:
1. I still need to modify the Makefile to change ninja to ninja-build.
2. The game didn't launch in full screen. (I know Antares runs as 800x600 or something similar, but would be nice if it auto-scaled the display)

If you like I can open those up a separate bug reports though.

Thanks Sfiera! 
I didn't see any place to donate to this project. I generally like to give something to open source projects that I find useful (or in this case fun!).

I'll see about making a spec file for this soon.

**Comment by sfiera** [comment]
1. Fixed at HEAD
2. It should scale up on demand. On Mac, the maximize button works. Does it not work/not work as well for you?

**Comment by Jeryia** [comment]
Thanks, 
I've tested and it's safe to say that Fedora now works with the given instructions. 

Maximize does work, though it's slightly annoying to see the title bar at the top of the screen. That should be a feature request in another ticket. 

Closing this ticket. I think we're set :+1:

**Comment by sfiera** [comment]
Cool, opened #167 for that (though I'm going to call it a bug, not a feature request, because hiding is the intended behavior and it works that way on Mac).

**Comment by Jeryia** [comment]
I've modified Mailaender's spec file to work with Fedora (Thanks Mailaender!). I've attached an rpm and a spec file for anyone who wants to use them.

Though I know we're not exactly on version 0.8.0, I've left the spec file at that version until a new release it done.

SPEC: http://www.mediafire.com/download/3xe95c87hiie3xi/antares-FC23.spec
RPM: http://www.mediafire.com/download/tqk1j9s2pgqex3d/antares-0.8.0-1.x86_64.rpm
